### PR TITLE
Add MBRBlockDevice API reference doc

### PR DESCRIPTION
**Pull request by kegilbert**
@AnotherButler 
@sg-

The first example also uses HeapBlockDevice, should I cross reference this doc with HeapBlockDevice?

### CODE REVIEW

**Comment by AnotherButler** [comment]
I think that would be a good idea.

File: docs/reference/api/storage/MBRBlockDevice.md [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/v5.6/mbed-os-api-doxy/class_m_b_r_block_device.html)
+
+### MBRBlockDevice example
+
+Partition a heap backed block device into two partitions.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/MBRBlockDevice_ex_1/)](https://os.mbed.com/teams/mbed_example/code/MBRBlockDevice_ex_1/file/daa62d7aa9f9/main.cpp)
+
+Partition an SD card, and format the new partition with a FAT filesystem. The SD card will now be recognized when plugged into a PC.
```
Comment (AnotherButler) [inline docs/reference/api/storage/MBRBlockDevice.md]: Query: Who or what now recognizes the SD card when you plug it into a PC? The PC? A specific program? Please clarify.

File: docs/reference/api/storage/MBRBlockDevice.md [hunk 2]
```diff
@@ -1,2 +1,17 @@
 ## MBRBlockDevice
 
+The MBRBlockDevice class provides a way to manage a Master Boot Record (MBR) on a storage device allowing for partitioning and formatting. The operating systems use the MBR to load the storage device properly into memory.
```
Comment (AnotherButler) [inline docs/reference/api/storage/MBRBlockDevice.md]: Query: To make sure I'm reading this correctly, is the first sentence saying that without the ability to manage an MBR, you can't partition and format? Also, what can't you partition and format?

**Comment by kegilbert** [comment]
@AnotherButler Rephrased the second example and intro. Also linked the first example to the HeapBlockDevice API page. 

To answer your first question (rephrased the intro to address this a bit better):  The MBR is not required to format a device, but will allow for partitioning the device.

**Comment by AnotherButler** [comment]
@kegilbert Nice. I did one last copy edit. It's good from a grammar perspective. Once the engineering review is complete and there are no new grammar edits, I'll merge this.

**Comment by kegilbert** [comment]
Mentioned in the docs meeting external links are OK, the header file had a link to the MBR wikipedia page so I added that for additional background information.

**Comment by AnotherButler** [comment]
@sg- Is this good to merge?

**Pull request merged**